core/domain/html_validation_service.py

```diff
@@ -705,3 +705,156 @@ def convert_svg_diagram_tags_to_image_tags(html_string):
         tabs['tab_contents-with-value'] = escape_html(
             json.dumps(tab_content_list))
     return python_utils.UNICODE(soup)
+
+def fix_incorrectly_encoded_chars(html_string):
+    """Replaces incorrectly encoded character with the correct one in a given
+    HTML string.
+
+    Args:
+        html_string: str. The HTML string to modify.
+
+    Returns:
+        str. The updated html string.
+    """
+    soup = bs4.BeautifulSoup(
+        html_string.encode(encoding='utf-8'), 'html.parser')
+    fixed_html = python_utils.UNICODE(soup)
+    char_mapping_tuples = [
```

Comment:
FYI, the human readable values can be found here: https://docs.google.com/spreadsheets/d/1xE_AKcWKXXoFXd2bAowrH6dQndnsyQsLaDKnnV98UHE/edit?usp=sharing